packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -1114,11 +1163,19 @@ export const Table = <
           {data.length !== 0 && (
             <>
               {enableVirtualization ? (
-                <VirtualScroll
-                  itemsLength={page.length}
-                  itemRenderer={virtualizedItemRenderer}
-                  scrollToIndex={scrollToIndex}
-                />
+                <div {...outerProps}>
```

Comment:
I'm thinking we should move this outer wrapper div into a `<ShadowRoot>` (attached to the table-body), so that it stays "hidden" from the light DOM.

Reply:
Added `<ShadowRoot>` in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/b6afd1397941969e87f9f3911001c6c2e32e842e).

Reply:
This doesn't do anything, because your wrapper div is not inside the shadow root.

I'm suggesting something like this:

```jsx
<ShadowRoot>
  {enableVirtualization ? (
    <div {...outerProps}><slot /></div>
  ) : (
    <slot />
  )}
</ShadowRoot>

{data.length !== 0 && (
  {enableVirtualization ? (
    virtualizer
      .getVirtualItems()
      .map((virtualItem) =>
        virtualizedItemRenderer(
          virtualItem.index,
          virtualItem,
          virtualizer,
        ),
      )
  ) : (
    page.map((_, index) => getPreparedRow(index))
  )}
)}
```

Reply:
Fixed in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/65c2229e35592be539d44d33cc0f9e923dd4f6fb).